### PR TITLE
Add Presto inverse_laplace_cdf function

### PR DESCRIPTION
**Pull request by jobbine-joseph**
implementation of inverse laplace cdf

### CODE REVIEW

**Comment by yzhang1991** [comment]
@aditi-pandit @majetideepak

File: velox/docs/functions/presto/math.rst [hunk 1]
```diff
@@ -55,6 +55,13 @@ Mathematical Functions
 
     Returns the value of ``string`` interpreted as a base-``radix`` number. ``radix`` must be between 2 and 36.
 
+.. function:: inverse_laplace_cdf(mean, scale, p) → double
```
Comment (aditi-pandit) [inline velox/docs/functions/presto/math.rst:58]: This function has to be documented with the rest of the inverse probability functions in Probability Functions : inverse_cdf portion of this file.

File: velox/functions/prestosql/Probability.h [hunk 2]
```diff
@@ -91,5 +92,29 @@ struct BinomialCDFFunction {
   }
 };
 
+template <typename T>
+struct InverseLaplaceCDFFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+  FOLLY_ALWAYS_INLINE void
+  call(double& result, double location, double scale, double p) {
+    if (std::isnan(location) || std::isnan(scale) || std::isnan(p)) {
```
Comment (aditi-pandit) [inline velox/functions/prestosql/Probability.h:100]: This code would be easier to read if written as 
if(std::isnan(...) {
  result = ...
  return;
}

if (scale....) {
  result = ....
  return;
}

without the nested ifs.

File: velox/functions/prestosql/Probability.h [hunk 3]
```diff
@@ -91,5 +92,29 @@ struct BinomialCDFFunction {
...
+      result = std::numeric_limits<double>::quiet_NaN();
+    } else if (scale <= 0 || p < 0 || p > 1) {
+      result = std::numeric_limits<double>::quiet_NaN();
+      VELOX_USER_CHECK_GT(scale, 0, "scale must be greater than 0");
```
Comment (aditi-pandit) [inline velox/functions/prestosql/Probability.h:104]: Its strange that these CHECKS happen after the assignment of the result. I think they are meant for the cases in the else part of the condition below.

Please can you check.

File: velox/functions/prestosql/Probability.h [hunk 4]
```diff
@@ -91,5 +92,29 @@ struct BinomialCDFFunction {
...
+      VELOX_USER_CHECK_GE(
```
Comment (aditi-pandit) [inline velox/functions/prestosql/Probability.h:105]: Please use the same error messages as the  java code.

Java code uses error "p must be in the interval [0,1]"

File: velox/functions/prestosql/tests/ProbabilityTest.cpp [hunk 5]
```diff
@@ -155,5 +155,37 @@ TEST_F(ProbabilityTest, binomialCDF) {
...
+TEST_F(ProbabilityTest, inverseLaplaceCDF) {
+  const auto inverseLaplaceCDF = [&](std::optional<double> location,
+                                     std::optional<double> scale,
+                                     std::optional<double> probability) {
+    return evaluateOnce<double>(
+        "inverse_laplace_cdf(c0, c1, c2)", location, scale, probability);
+  };
+
+  // Test cases
+  EXPECT_DOUBLE_EQ(0.0, inverseLaplaceCDF(0.0, 1.0, 0.5).value());
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ProbabilityTest.cpp:167]: Use EXPECT_EQ here.

File: velox/functions/prestosql/tests/ProbabilityTest.cpp [hunk 6]
```diff
@@ -155,5 +155,37 @@ TEST_F(ProbabilityTest, binomialCDF) {
...
 
+TEST_F(ProbabilityTest, inverseLaplaceCDF) {
+  const auto inverseLaplaceCDF = [&](std::optional<double> location,
+                                     std::optional<double> scale,
+                                     std::optional<double> probability) {
+    return evaluateOnce<double>(
+        "inverse_laplace_cdf(c0, c1, c2)", location, scale, probability);
+  };
+
+  // Test cases
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ProbabilityTest.cpp:166]: Remove this comment.

File: velox/functions/prestosql/tests/ProbabilityTest.cpp [hunk 7]
```diff
@@ -155,5 +155,37 @@ TEST_F(ProbabilityTest, binomialCDF) {
...
+  EXPECT_DOUBLE_EQ(0.0, inverseLaplaceCDF(0.0, 1.0, 0.5).value());
+  EXPECT_DOUBLE_EQ(5.0, inverseLaplaceCDF(5.0, 2.0, 0.5).value());
+  // Test NaN inputs
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ProbabilityTest.cpp:169]: Remove this comment.

File: velox/functions/prestosql/tests/ProbabilityTest.cpp [hunk 8]
```diff
@@ -155,5 +155,37 @@ TEST_F(ProbabilityTest, binomialCDF) {
...
+      inverseLaplaceCDF(1.0, 1.0, std::numeric_limits<double>::quiet_NaN()),
+      IsNan());
+  EXPECT_THAT(
+      inverseLaplaceCDF(
+          std::numeric_limits<double>::infinity(),
+          1.0,
+          std::numeric_limits<double>::quiet_NaN()),
+      IsNan());
+
+  // Test invalid scale
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ProbabilityTest.cpp:183]: Remove this comment.

File: velox/functions/prestosql/tests/ProbabilityTest.cpp [hunk 9]
```diff
@@ -155,5 +155,37 @@ TEST_F(ProbabilityTest, binomialCDF) {
       binomialCDF(-2, 0.5, -1), "numberOfTrials must be greater than 0");
 }
 
+TEST_F(ProbabilityTest, inverseLaplaceCDF) {
+  const auto inverseLaplaceCDF = [&](std::optional<double> location,
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ProbabilityTest.cpp:159]: Please add test cases with std::nullopt, kDoubleMin, kDoubleMax, kInf

**Comment by yzhang1991** [comment]
Hi @jobbine-joseph, just in case you are not aware, we are holding on merging any inverse function implementations for now due to this issue https://github.com/facebookincubator/velox/pull/5382#issuecomment-1651920130. We are waiting for a fix from Meta.
CC @aditi-pandit

**Comment by jobbine-joseph** [comment]
@aditi-pandit can you review this ?
Thank you

**Pull request closed without merging**